homeassistant/components/ping/sensor.py

```diff
@@ -0,0 +1,117 @@
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DURATION,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda result: result.data.get("avg"),
+        has_fn=lambda result: "avg" in result.data,
+    ),
+    PingSensorEntityDescription(
+        key="round_trip_time_max",
+        translation_key="round_trip_time_max",
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DURATION,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda result: result.data.get("max"),
+        has_fn=lambda result: "max" in result.data,
+    ),
+    PingSensorEntityDescription(
+        key="round_trip_time_mdev",
+        translation_key="round_trip_time_mdev",
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DURATION,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda result: result.data.get("mdev"),
+        has_fn=lambda result: "mdev" in result.data,
+    ),
+    PingSensorEntityDescription(
+        key="round_trip_time_min",
+        translation_key="round_trip_time_min",
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.DURATION,
+        entity_registry_enabled_default=False,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda result: result.data.get("min"),
+        has_fn=lambda result: "min" in result.data,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Ping sensors from config entry."""
+    data: PingDomainData = hass.data[DOMAIN]
+    coordinator = data.coordinators[entry.entry_id]
+
+    async_add_entities(
+        PingSensor(entry, description, coordinator)
+        for description in SENSORS
+        if description.has_fn(coordinator.data)
+    )
+
+
+class PingSensor(PingEntity, SensorEntity):
+    """Represents a Ping sensor."""
+
+    entity_description: PingSensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        description: PingSensorEntityDescription,
+        coordinator: PingUpdateCoordinator,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, f"{config_entry.entry_id}-{description.key}")
+
+        self.entity_description = description
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.coordinator.data.is_alive
```

Comment:
```suggestion
        return self.coordinator.data.is_alive
```
Maybe add also `super().available` so it's also unavailable if coordinator can't fetch data